jschon/jsonpointer.py

```diff
@@ -70,6 +70,12 @@ class JSONPointer(Sequence[str]):
        and referred to as *keys* in the JSONPointer class.
     """
 
+    malformed_exc: Type[JSONPointerMalformedError] = JSONPointerMalformedError
```

Comment:
😮 I'm rather shocked I was able to run the code.  Type hinting still baffles me a bit.

Reply:
As I understand it, type hints are not evaluated at runtime... in ordinary people's code. In magical code, as in Pydantic, type hints cause magical validation to occur at runtime.

As an (non-magical) example, the `jsonpointer` module has the lines:
```py
if TYPE_CHECKING:
    from jschon.json import JSON
```
 So the import is not even executed at runtime. It's just used by typecheckers and IDEs. This is fine when `JSON` only appears in type annotations. But if it appears in runtime code, as in `JSONPointer.evaluate()`:
```py
                if (isjson := isinstance(value, JSON)) and value.type == "object" or \
```
then it must be imported in runtime code, so to avoid a circular import we import `JSON` locally in that method.

Reply:
Thanks for the explanation!